code/modules/mob/living/carbon/xenomorph/strains/castes/hivelord/designer.dm

```diff
@@ -0,0 +1,404 @@
+		qdel(existing_cost_node)
+		new /obj/effect/alien/weeds(target_turf)
+		xeno.visible_message(SPAN_XENODANGER("[xeno] reverts the node into weeds."), \
+		SPAN_XENONOTICE("We restructure the node, reverting it into weeds."), null, 5)
+	else
+		// Create a new cost node normally
+		xeno.visible_message(SPAN_XENODANGER("[xeno] surges the resin, creating a strange looking node!"), \
+		SPAN_XENONOTICE("We create a flexible node!"), null, 5)
+		var/cost_nodes = new /obj/effect/alien/weeds/node/designer/cost(target_turf)
+		xeno.cost_node_list += cost_nodes
+		playsound(target_turf, "alien_resin_build", 25)
+
+		if(length(xeno.cost_node_list) > max_cost_nodes)
+			addtimer(CALLBACK(src, "remove_oldest_cost_node", xeno), 0)
+
+	xeno_cooldown = COOLDOWN_MULTIPLIER
+	REMOVE_TRAIT(owner, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("design_cost"))
+	apply_cooldown()
+	xeno_cooldown = initial(xeno_cooldown)
+
+	REMOVE_TRAIT(owner, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("design_cost"))
+	return ..()
+
+/datum/action/xeno_action/activable/design_cost_node/proc/remove_oldest_cost_node(mob/living/carbon/xenomorph/xeno)
+	if(length(xeno.cost_node_list) > 0)
+		var/obj/effect/alien/weeds/node/designer/cost/oldest_cost_node = xeno.cost_node_list[1]
+		if(oldest_cost_node)
+			var/turf/old_cost_loc = get_turf(oldest_cost_node.loc) // Get the turf of the oldest node
+			if(old_cost_loc) // Ensure the turf exists
+				new /obj/effect/alien/weeds(old_cost_loc) // Replace with a new /obj/effect/alien/weeds
+			qdel(oldest_cost_node) // Safely delete the old node
+		xeno.cost_node_list.Cut(1, 2) // Remove the first element from the list
+
+//////////////////////////
+// Greater Resin Surge. //
+//////////////////////////
+
+/datum/action/xeno_action/verb/verb_greater_surge()
+	set category = "Alien"
+	set name = "Greater Resin Surge"
+	set hidden = TRUE
+	var/action_name = "Greater Resin Surge"
+	handle_xeno_macro(src, action_name)
+
+/datum/action/xeno_action/activable/greater_resin_surge
+	name = "Greater Resin Surge (250)"
+	action_icon_state = "greater_resin_surge"
+	plasma_cost = 250
+	xeno_cooldown = 30 SECONDS
+	macro_path = /datum/action/xeno_action/verb/verb_greater_surge
+	action_type = XENO_ACTION_CLICK
+	ability_primacy = XENO_PRIMARY_ACTION_5
+
+/datum/action/xeno_action/activable/greater_resin_surge/use_ability(atom/target_atom)
+	var/mob/living/carbon/xenomorph/xeno = owner
+	if(!istype(xeno))
+		return
+
+	if(!action_cooldown_check())
+		return
+
+	if(!xeno.check_state(TRUE))
+		return
+
+	if(!do_after(xeno, DO_AFTER_DELAY, INTERRUPT_ALL, BUSY_ICON_HOSTILE))
+		return
+
+	if(!check_and_use_plasma_owner())
+		return
+
+	// Create overlays for speed nodes and schedule their replacement
+	for(var/obj/effect/alien/weeds/node/designer/speed/node in xeno.speed_node_list)
+		if(node)
+			var/turf/node_loc = get_turf(node.loc)
+			if(node_loc)
+				create_animation_overlay(node_loc, /obj/effect/resin_construct/fastweak)
+
```

Comment:
you dont need to check if node is there because `for` already has istype check in it. You could even place `as anything` in the `for` loop, because you already know only speed nodes will be there.